File: C#-Server+Client/BaseLibrary/communication/command/repairman/RepairAnswerCommand.cs
using System;
using System.Collections.Generic;

namespace BaseLibrary.communication.command.repairman {
    public class RepairAnswerCommand : ARepairmanCommand {

        private static readonly List<ISubCommandFactory> SUB_COMMAND_FACTORIES = new List<ISubCommandFactory>();

        public static int RegisterSubCommandFactory(ISubCommandFactory subCommandFactory) {
            int position = SUB_COMMAND_FACTORIES.Count;
            SUB_COMMAND_FACTORIES.Add(subCommandFactory);
            return position;
        }

        private bool _success;
        /// <summary>
        /// <list type="bullet">
        ///   <item>
        ///      <description>true - robot use repair tool to repair robots in range.</description>
        ///   </item>
        ///   <item>
        ///      <description>false - robot do not use repair tool to repair. Robot use repair tool too many times.</description>
        ///   </item>
        /// </list>
        /// </summary>
        public bool SUCCESS {
            get {
                if (pending)
                    throw new NotSupportedException("Cannot access to property of pending request.");
                return _success;
            } private set => _success = value; }

        /// <summary>
        /// Using for create pending command.
        /// </summary>
        public RepairAnswerCommand() { }

        /// <summary>
        /// Using for create command fulled with data.
        /// </summary>
        public RepairAnswerCommand(bool success) {
            SUCCESS = success;
            pending = false;
        }

        /// <summary>
        /// Fill command by data from another command (use full for filling pending command by command with data).
        /// </summary>
        /// <param name="source"></param>
        public void FillData(RepairAnswerCommand source) {
            SUCCESS = source.SUCCESS;
            pending = false;
        }
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/communication/command/repairman/RepairAnswerCommand.java

package BaseLibrary.communication.command.repairman;

import BaseLibrary.communication.protocol.AProtocol;

public class RepairAnswerCommand extends ARepairmanCommand {

	private static final String NAME = "REPAIR_ANSWER";

	static {
		AProtocol.registerForDeserialize(NAME, RepairAnswerCommand.class);
	}

	@Override
	public String getCommandName() {
		return NAME;
	}
	
	private boolean success;

	public RepairAnswerCommand() {
	}

	public RepairAnswerCommand(boolean success) {
		setSuccess(success);
	}

	/**
	 * Fill command by data from another command (use-full for filling pending
	 * command by command with data).
	 */
	public void FillData(RepairAnswerCommand source) {
		setSuccess(source.success);
	}

	/**
	 * <ul>
	 * <li><description>true - robot use repair tool to repair robots in
	 * range.</description></li>
	 * <li><description>false - robot do not use repair tool to repair. Robot use
	 * repair tool too many times.</description></li>
	 * </ul>
	 */
	public boolean isSuccess() {
		if (pending)
			throw new UnsupportedOperationException("Cannot access to property of pending request.");
		return success;
	}

	private void setSuccess(boolean success) {
		pending = false;
		this.success = success;
	}
}
